connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -571,7 +564,7 @@ class DataFrameReader private[sql] (sparkSession: SparkSession) extends Logging
   // Builder pattern config options
   ///////////////////////////////////////////////////////////////////////////////////////
 
-  private var source: String = _
+  private var source: Option[String] = None
```

Comment:
This is a nice code quality improvement, but why are we making this improvement in the 3.5 branch but not in the master branch?

Reply:
The improvement for the master branch is in a separate PR: https://github.com/apache/spark/pull/51761. It may be worth checking whether we need to backport it to the 4.0 branch.